### PR TITLE
[lit-labs/compiler] gracefully handle bindings in tag name position

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
Issue: https://github.com/lit/lit/issues/189
RFC: https://github.com/lit/rfcs/pull/21
Full prototype PR: https://github.com/lit/lit/pull/3984

### Why

In DEV mode, this results in the error: `Bindings in tag names are not supported`.

### How

If a marker exists in the tag name then we know that the user supplied a binding in tag name position, and can stop compiling the template.

### Test plan

Tested by adding a golden.

### Alternative

We could add a diagnostic here – however I've opted for a simpler but correct solution as there is no diagnostic support wired into the compiler, and this is one of the last cases we need to handle.

### CODE REVIEW

File: packages/labs/compiler/src/lib/template-transform.ts [hunk 1]
```diff
@@ -283,6 +283,13 @@ class CompiledTemplatePass {
       'pre:node': (node): boolean | void => {
         if (isElementNode(node)) {
           const attributesToRemove = new Set<unknown>();
+          if (node.tagName.includes(marker)) {
+            // Do not compile template if a marker was inserted in tag name
+            // position.
+            // TODO: Provide a diagnostic here.
+            shouldCompile = false;
+            return true;
```
Comment (augustjk) [inline packages/labs/compiler/src/lib/template-transform.ts:291]: Is there value to returning `true` here? There's no need to traverse into the children if we're not compiling, right?

File: packages/labs/compiler/src/lib/template-transform.ts [hunk 2]
```diff
@@ -283,6 +283,13 @@ class CompiledTemplatePass {
       'pre:node': (node): boolean | void => {
         if (isElementNode(node)) {
           const attributesToRemove = new Set<unknown>();
+          if (node.tagName.includes(marker)) {
+            // Do not compile template if a marker was inserted in tag name
+            // position.
+            // TODO: Provide a diagnostic here.
```
Comment (augustjk) [inline packages/labs/compiler/src/lib/template-transform.ts:289]: Nit. I think we like to add a username to TODOs so we know who might have context.

File: packages/labs/compiler/src/lib/template-transform.ts (hunk 1 above)
Comment (AndrewJakubowicz) [reply packages/labs/compiler/src/lib/template-transform.ts:291]: Whoops! This should be false.

File: packages/labs/compiler/src/lib/template-transform.ts (hunk 2 above)
Comment (AndrewJakubowicz) [reply packages/labs/compiler/src/lib/template-transform.ts]: Agree – done

**Review by rictic — APPROVED** [review]

**Pull request merged**